qiskit_quantuminspire/qi_jobs.py

```diff
@@ -220,7 +221,26 @@ def result(self) -> Result:
 
     def status(self) -> JobStatus:
         """Return the status of the (batch)job, among the values of ``JobStatus``."""
-        return JobStatus.DONE
+
+        # mapping of QI2 BatchJobStatus to Qiskit JobStatus
+        status_map = {
+            BatchJobStatus.QUEUED: JobStatus.QUEUED,
+            BatchJobStatus.RESERVED: JobStatus.QUEUED,
+            BatchJobStatus.PLANNED: JobStatus.QUEUED,
+            BatchJobStatus.RUNNING: JobStatus.RUNNING,
+            BatchJobStatus.FINISHED: JobStatus.DONE,
+        }
+
+        batch_job = asyncio.run(self._fetch_batchjob_status())
+        return status_map[batch_job.status]
+
+    async def _fetch_batchjob_status(self) -> BatchJob:
+        async with ApiClient(config()) as api_client:
+            api_instance = BatchJobsApi(api_client)
+            batchsjob_page = await api_instance.read_batch_jobs_batch_jobs_get(id=self.batch_job_id)
+            if not batchsjob_page.items:
+                raise RuntimeError(f"No (batch)job in platform with id {self.batch_job_id}")
+            return batchsjob_page.items[0]
```

Comment:
We should be avoiding manually fiddling around in pages (what you're doing here will be useful elsewhere too and before you know it this bit of code is duplicated all over the place). My proposal would be to implement a function in PageReader that gets a single item from an endpoint that returns pages (which does essentially the above and also checks if the endpoint returned exactly 1 item).

The fact that such a thing is necessary at all may say something about the way we structured our api. Having some endpoints that return specific items instead of lists/pages would be convenient. But that is a discussion for another time.

Reply:
Added the method ` get_single `to the PageReader helper class